packages/bits/src/services/unit-conversion.service.ts

```diff
@@ -72,11 +78,25 @@ export class UnitConversionService {
     public getFullDisplay(conversion: IUnitConversionResult, unit: UnitOption, plusSign = false, nanDisplay = "---"): string {
         const isValidNumber = this.isValidNumber(conversion.value);
         const spacing = unit !== "generic" && isValidNumber ? " " : "";
-        const unitDisplay = isValidNumber ? this.getUnitDisplay(conversion, unit) : "";
+        let unitDisplay = isValidNumber ? this.getUnitDisplay(conversion, unit) : "";
 
         // The generic unit is not currently i18n friendly
         const localizeValue = unit !== "generic";
-        return `${this.getValueDisplay(conversion, plusSign, nanDisplay, localizeValue)}${spacing}${unitDisplay}`;
+
+        let displayValue: string;
+
+        if (!unitDisplay && !(conversion.value === "NaN") && conversion.order ) {
```

Comment:
This might be a bit clearer to read:

```suggestion
        if (!unitDisplay && conversion.value !== "NaN" && conversion.order ) {
```